shared/ssl.sh

```diff
@@ -64,9 +72,31 @@ CERTSUBJECT="/CN=${CERTNAME}"
 CERTHEADER="-----BEGIN CERTIFICATE-----"
 CURLFLAGS="--silent --show-error --cacert ${CACERTFILE} --retry 5 --retry-connrefused --retry-delay 2"
 
+### Handle certificate extensions
+# NOTE If we want to expand support for more extensions, it would be better
+# to define them in a .conf file rather than directly on the CLI.
+# That would also work on older versions of openssl that don't support
+# the `-addext` flag.
+# For now, we explicitly handle DNS alt names because it's simpler.
+CERTEXTENSIONS=""
+if [ -n "${DNS_ALT_NAMES}" ]; then
+    NAMES=""
+    first=true
+    for name in ${DNS_ALT_NAMES}; do
```

Comment:
I can't find a simple way to iterate over a **comma-separated** string in `sh` without doing things with `sed` or messing with the `IFS`, or something equally fragile.

For this reason I went with a **space-separated** value for the `DNS_ALT_NAMES`. 

I also verified we can correctly and easily specify space-separated environment variables in a docker-compose YAML file.

Reply:
Ok turns out we can easily use `tr "," " "` here to support comma-supported, which works on alpine, centos, ubuntu.